app/controllers/tools_controller.rb

```diff
@@ -0,0 +1,79 @@
+class ToolsController < ApplicationController
+  require 'saml_idp'
+
+  def new
+    @valid_request = ""
+    @valid_signature = ""
+    @matching_cert_sn = ""
+    @xml = ""
+  end
+
+  def index
+    flash[:error] = nil
+
+    return if !auth_params
+
+    sp = ServiceProvider.find_by(issuer: auth_request.issuer)
+
+    begin
+      certs = (get_cert_param || sp.certs).map { |cert| OpenSSL::X509::Certificate.new(cert) }
+    rescue OpenSSL::X509::CertificateError
+      flash[:error] = "Something is wrong with the certificate you submitted."
+    rescue NoMethodError
+      flash[:error] = "Could not find any certificates to use. Please add a certificate to your application configuration or paste one below."
```

Comment:
I feel like it would be clearer to check `certs.empty?` rather than rely on a NoMethodError for this particular error message, `NoMethodError` can come from anywhere